lib/veri_helpers.ml

```diff
@@ -0,0 +1,45 @@
+open Core_kernel.Std
+open Bap.Std
+open Bap_traces.Std
+
+module Dis  = Disasm_expert.Basic
+
+type insn = Dis.full_insn
+
+let lift_insn arch (mem,insn) = 
+  let module Target = (val target_of_arch arch) in
+  match Target.lift mem insn with
+  | Ok bil -> Some bil
+  | Error _ -> None
+
+let insn_of_mem dis mem =
+  let open Or_error in
+  Dis.insn_of_mem dis mem >>= (fun (imem, insn, left) -> 
+      match insn with 
+      | Some insn -> return (imem, insn)
+      | None -> error_string "no instruction in memory")
+
+let insn_of_chunk arch chunk =
+  let open Or_error in
+  let endian = Arch.endian arch in
+  Dis.with_disasm ~backend:"llvm" (Arch.to_string arch)
+    ~f:(fun dis ->
+        let dis = Dis.store_kinds dis |> Dis.store_asm in
+        let mems = Bigstring.of_string (Chunk.data chunk) in
+        Memory.create endian (Chunk.addr chunk) mems >>=
+        fun mem -> insn_of_mem dis mem)
+
+let mem_of_arch arch = 
+  let (module T : Target) = target_of_arch arch in
+  T.CPU.mem
+
+let memory_operands mv = 
+  let width = Word.bitwidth (Move.data mv) in
+  Size.of_int_opt width |> function
+  | None -> 
+    let er = Printf.sprintf "wrong size for memory operand: %d" width in
+    failwith er
```

Comment:
wow, no exceptions please. The only possible case for using `failwith` is when you're reporting a programmer error. The result of disassembly is expected to have errors (that's why we're implementing `veri`). The error should be reported.